distogram/__init__.py

```diff
@@ -373,57 +335,62 @@ def histogram(h, ucount=100):
         bins in the Distogram object.
-    '''
+    """
     if len(h.bins) < h.bin_count:
         raise ValueError("not enough elements in distribution")
 
-    last = 0.0
-    u = []
-    bounds = _linspace(h.min, h.max, num=ucount+1)
-    for e in bounds[1:-1]:
-        new = count_at(h, e)
-        u.append((e, new-last))
-        last = new
+    bin_bounds = _linspace(h.min, h.max, num=ucount+1)
+    counts = [count_at(h, e) for e in bin_bounds[1:-1]]
+    u = [(bin_bounds[0], counts[0])]
+    u.extend([
+        (b, new - last)
+        for b, new, last in zip(bin_bounds[1:], counts[1:], counts[:-1])
+    ])
+
     return u
 
 
-def quantile(h, value):
-    '''Returns a quantile of the distribution
+def quantile(h: Distogram, value: float) -> Optional[float]:
+    """ Returns a quantile of the distribution
 
     Args:
         h: A Distogram object.
         value: The quantile to compute. Must be between 0 and 1
 
     Returns:
         An estimation of the quantile. Returns None if the Distogram
-        object contains no element or value is outside of [0:1].
-    '''
+        object contains no element or value is outside of (0:1).
+    """
     if len(h.bins) == 0:
         return None
 
-    if value < 0.0 or value > 1.0:
+    if not (0.0 < value < 1.0):
         return None
 
     total_count = count(h)
     q_count = int(total_count * value)
-    bins = h.bins
-    i = 0
-    if q_count <= (bins[0][1] / 2):  # left values
-        print("left")
-        ratio = q_count / (bins[0][1] / 2)
-        value = h.min + (ratio * (bins[0][0] - h.min))
-    elif q_count >= (total_count - (bins[-1][1] / 2)):  # right values
-        base = q_count - (total_count - (bins[-1][1] / 2))
-        ratio = base / (bins[-1][1] / 2)
-        value = bins[-1][0] + (ratio * (h.max - bins[-1][0]))
-        print("right value for quantile base: {}, ratio: {}".format(
-            base, ratio))
+    v0, f0 = h.bins[0]
+    vl, fl = h.bins[-1]
+
+    if q_count <= (f0 / 2):  # left values
+        fraction = q_count / (f0 / 2)
+        result = h.min + (fraction * (v0 - h.min))
+
+    elif q_count >= (total_count - (fl / 2)):  # right values
+        base = q_count - (total_count - (fl / 2))
+        fraction = base / (fl / 2)
+        result = vl + (fraction * (h.max - vl))
+
     else:
-        mb = q_count - bins[0][1] / 2
-        while mb - (bins[i][1] + bins[i+1][1]) / 2 > 0 and i < len(bins) - 1:
-            mb -= (bins[i][1] + bins[i+1][1]) / 2
-            i += 1
-
-        ratio = mb / ((bins[i][1] + bins[i+1][1]) / 2)
-        value = bins[i][0] + (ratio * (bins[i+1][0] - bins[i][0]))
-    return value
+        mb = q_count - f0 / 2
+        vi, fi, vj, fj = 0, 0, 0, 0
+        for (vi, fi), (vj, fj) in zip(h.bins[:-1], h.bins[1:]):
+            if mb > (temp := (fi + fj) / 2):
+                mb -= temp
+            else:
+                break
+
+        fraction = mb / ((fi + fj) / 2)
+        result = vi * (1 - fraction) + vj * fraction
```

Comment:
While the conversion is correct, it may be harder understand its meaning/reason when reading the code. i'd prefer to keep the original notation:
```
result = vi + (fraction * (vj - vi))
```

Reply:
done